kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt

```diff
@@ -1,72 +1,1061 @@
+              onDoubleRange { double, plusMinusValue ->
+                double.shouldMatchBetween(double - plusMinusValue, double + plusMinusValue, 0.0)
+                
+              }
             }
+          }
+          
+          "When it's equal to the last number of the range" - {
+            
+            "With tolerance" {
+              onDoubleRange { double, _ ->
+                double.shouldMatchBetween(double - delta, double, delta)
+              }
+            }
+            
+            "Without tolerance" {
+              onDoubleRange { double, _ ->
+                double.shouldMatchBetween(double - delta, double, 0.0)
+              }
+            }
+          }
+        }
+        
+        "Should not match between" - {
+          
+          "When it's smaller than the first number of the range" - {
+            
+            "With tolerance" {
+              onDoubleRange { double, valueToVariate ->
+                double.shouldNotMatchBetween(double + valueToVariate, double + 2 * valueToVariate, delta)
+              }
+            }
+            
+            "Without tolerance" {
+              onDoubleRange { double, valueToVariate ->
+                double.shouldNotMatchBetween(double + valueToVariate, double + 2 * valueToVariate, 0.0)
+              }
+            }
+          }
+          
+          "When it's bigger than the last number of the range" - {
+            
+            "With tolerance" {
+              onDoubleRange { double, valueToVariate ->
+                double.shouldNotMatchBetween(double - 2 * valueToVariate, double - valueToVariate, delta)
+              }
+            }
+            
+            "Without tolerance" {
+              onDoubleRange { double, valueToVariate ->
+                double.shouldNotMatchBetween(double - 2 * valueToVariate, double - valueToVariate, 0.0)
+              }
+            }
+          }
+        }
+        
+        
+      }
+    }
+    
+    "Less than matcher" - {
+      
+      "Every numeric double" - {
+        
+        "Should be less than" - {
+          
+          "Numbers bigger than itself" {
+            onDoubleRange { double, valueToVariate ->
+              double shouldMatchLessThan double + delta
+              double shouldMatchLessThan double + valueToVariate
+            }
+          }
+          
+          "Infinity" {
+            onDoubleRange { double, _ ->
+              double shouldMatchLessThan POSITIVE_INFINITY
+            }
+          }
+        }
+        
+        
+        "Should not be less than" - {
+          
+          "Itself" {
+            onDoubleRange { double, _ ->
+              double shouldNotMatchLessThan double
+            }
+          }
+          
+          "Numbers smaller than itself" {
+            onDoubleRange { double, valueToVariate ->
+              double shouldNotMatchLessThan double - delta
+              double shouldNotMatchLessThan double - valueToVariate
```

Comment:
I have the same comment about `valueToVariate` that I did about `delta`: it's not valid if `double` has a large positive or negative exponent.

Reply:
I'm not quite sure if I understood this. Do you have suggestions on how to deal with this?